include/zephyr/drivers/interrupt_controller/riscv_plic.h

```diff
@@ -42,11 +92,27 @@ int riscv_plic_irq_is_enabled(uint32_t irq);
  */
 void riscv_plic_set_priority(uint32_t irq, uint32_t prio);
 
+/**
+ * @brief Set interrupt priority of an interrupt controller
+ *
+ * See @ref riscv_plic_set_priority for more info
+ *
+ * @param intc Interrupt controller of the irq, the first instance will be used if this is NULL
+ */
+void riscv_plic_intc_set_priority(const struct device *intc, uint32_t irq, uint32_t prio);
+
 /**
  * @brief Get active interrupt ID
  *
  * @return Returns the ID of an active interrupt
  */
 int riscv_plic_get_irq(void);
 
+/**
+ * @brief Get the interrupt controller of the active interrupt ID
+ *
+ * @return Returns the device struct of the interrupt controller
+ */
+const struct device *riscv_plic_get_dev(void);
```

Comment:
There can be multiple active interrupts on separate cores - is that taken into consideration here?

Reply:
SMP is not considered in this PR